### PR TITLE
Delete some unused code in run_dockerbuild.sh and Enable Nuget CUDA tests

### PR DESCRIPTION
**Pull request by snnn**
**Description**:

1. Remove some unused code and simplify tools/ci_build/github/linux/run_dockerbuild.sh. 
2. Enable Nuget CUDA tests. The original design was we could leverage [Directory.Build.props](https://docs.microsoft.com/en-us/visualstudio/msbuild/customize-your-build?view=vs-2019) and let cmake generate the required properties(USE_CUDA/...) there. However, in nuget packaging pipeline we test the package on a different host that doesn't run cmake command and doesn't have the auto-generated Directory.Build.props file.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Comment by snnn** [comment]
It has failures in nuget pipelines. I'm fixing them.

**Review by pranavsharma — APPROVED** [review]
Can you update the title of this PR reflecting the fact that you're also enabling GPU C# tests? Thanks!

File: csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest.sh [hunk 1]
```diff
@@ -29,7 +28,11 @@ if [ $RunTestCsharp = "true" ]; then
     exit 1
   fi
 
-  dotnet test $BUILD_SOURCESDIRECTORY/csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/Microsoft.ML.OnnxRuntime.EndToEndTests.csproj --no-restore --verbosity detailed
+  if [ $PACKAGENAME = "Microsoft.ML.OnnxRuntime.Gpu" ]; then
+    dotnet test -p:DefineConstants=USE_CUDA $BUILD_SOURCESDIRECTORY/csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/Microsoft.ML.OnnxRuntime.EndToEndTests.csproj --no-restore --verbosity detailed
```
Comment (snnn) [inline csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest.sh:32]: This line of change enables CUDA tests for the C# code.

File: tools/ci_build/github/linux/docker/scripts/install_ubuntu.sh [hunk 2]
```diff
@@ -3,12 +3,10 @@ set -e
 while getopts p:d: parameter_Option
 do case "${parameter_Option}"
 in
-p) PYTHON_VER=${OPTARG};;
```
Comment (baijumeswani) [inline tools/ci_build/github/linux/docker/scripts/install_ubuntu.sh:6]: Removing python version as input is causing some pipelines to fail. For example the orttraining-distributed pipeline is not able to find the python executable and failing as a result.

Comment (snnn) [reply tools/ci_build/github/linux/docker/scripts/install_ubuntu.sh:6]: Fixed in #8136

**Pull request merged**